src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -289,6 +343,15 @@ p5.RendererGL.prototype._drawImmediateFill = function() {
     this.immediateMode.shapeMode = constants.TRIANGLE_FAN;
   }
 
+  // WebGL 1 doesn't support the QUADS and QUAD_STRIP modes, so we
```

Comment:
This is likely why the QUADS shape mode was not added earlier. Great to have this fallback behavior available even if we do eventually offer webgl2 support (#2536 )